src/content-helper/dashboard-page/components/posts-table/components/single-post-row.tsx

```diff
@@ -0,0 +1,190 @@
+/**
+ * External dependencies
+ */
+import { Link } from 'react-router';
+
+/**
+ * WordPress dependencies
+ */
+import { DropdownMenu, MenuGroup, MenuItem, Spinner } from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { moreVertical } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { HydratedPost } from '../../../../common/providers/base-wordpress-provider';
+import { PostStats } from '../../../../common/providers/stats-provider';
+import { getPostEditUrl } from '../../../../common/utils/post';
+import { PostDetails } from './post-details';
+
+/**
+ * ActionDropdownProps type.
+ *
+ * @since 3.19.0
+ */
+type ActionDropdownProps = {
+	post: HydratedPost;
+};
+
+/**
+ * ActionDropdown component.
+ *
+ * Represents the action dropdown for each post in the PostsTable.
+ *
+ * @since 3.18.0
+ *
+ * @param {ActionDropdownProps} props The props for the ActionDropdown component.
+ */
+const ActionDropdown = ( { post }: ActionDropdownProps ): React.JSX.Element => (
+	<DropdownMenu icon={ moreVertical } label={ __( 'Actions', 'wp-parsely' ) }>
+		{ ( { onClose } ) => (
+			<MenuGroup>
+				<MenuItem
+					onClick={ () => {
+						window.open( post.link, '_blank', 'noopener,noreferrer' );
+						onClose();
+					} }
+				>
+					{ __( 'View', 'wp-parsely' ) }
+				</MenuItem>
+				<MenuItem
+					onClick={ () => {
+						window.open( getPostEditUrl( post.id ), '_blank', 'noopener,noreferrer' );
+						onClose();
+					} }
+				>
+					{ __( 'Edit', 'wp-parsely' ) }
+				</MenuItem>
+			</MenuGroup>
+		) }
+	</DropdownMenu>
+);
+
+/**
+ * SinglePostRowProps type.
+ *
+ * @since 3.18.0
+ */
+type SinglePostRowProps = {
+	post: HydratedPost;
+	index: number;
+	onPostClick?: ( post: HydratedPost ) => void;
+	stats?: PostStats;
+	compact: boolean;
+	isLoadingStats: boolean;
+	showStats?: boolean;
+	showActions?: boolean;
+	onErrorLoadingStats?: ( post: HydratedPost ) => void;
+};
+
+/**
+ * SinglePostRow component.
+ *
+ * Represents a single post row in the PostsTable.
+ *
+ * @since 3.18.0
+ *
+ * @param {SinglePostRowProps} props The props for the SinglePostRow component.
+ */
+export const SinglePostRow = ( {
+	post,
+	index,
+	onPostClick,
+	stats,
+	compact,
+	isLoadingStats: initialIsLoadingStats,
+	showStats = true,
+	showActions = true,
+	onErrorLoadingStats,
+}: SinglePostRowProps ): React.JSX.Element => {
+	const [ isLoadingStats, setIsLoadingStats ] = useState( initialIsLoadingStats );
+
+	const viewsWithoutCommas = stats?.views?.replace( /,/g, '' );
```

Comment:
Although this code should be OK, a more defensive method would be stripping out any non-numeric chars. There's an example of this [here](https://github.com/Parsely/wp-parsely/blob/59790dc0b56d0537a4ebb358cea8fd6b37e5c6a4/src/content-helper/common/utils/number.ts#L12), where we also directly convert to an integer.

```js
const number = parseInt( value.replace( /\D/g, '' ), 10 );
```

Reply:
Good improvement, I've added that in https://github.com/Parsely/wp-parsely/pull/3293/commits/b0f374ca7cf1152d8bd67a921a5b515abe14412d.